TestSQLStatsDataDriven is flaky

It looks like TestSQLStatsDataDriven occasionally flakes.   2 out of the last 1000 runs. 

<img width="883" alt="Screen Shot 2022-10-12 at 11 19 12 AM" src="https://user-images.githubusercontent.com/6658984/195444484-82811497-f6f6-48a3-a4c2-cd7cb158cb08.png">

Failed
=== RUN   TestSQLStatsDataDriven
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/83b49f90eb84fbc5d893648fc9fd7294/logTestSQLStatsDataDriven2913649257
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestSQLStatsDataDriven
    datadriven_test.go:223: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/83b49f90eb84fbc5d893648fc9fd7294/logTestSQLStatsDataDriven2913649257
--- FAIL: TestSQLStatsDataDriven (8.34s)

Its hard to tell whats going on here. But its flaked again recently.  I see this at the end of the log:

W221012 19:26:22.304027 2704 kv/kvserver/closedts/sidetransport/receiver.go:139 â‹® [n2] 264  closed timestamps side-transport connection dropped from node: 1
W221012 19:26:22.304076 1929 jobs/registry.go:824 â‹® [n2] 265  canceling all adopted jobs due to stopper quiescing
I221012 19:26:22.305595 2435 sql/stats/automatic_stats.go:509 â‹® [n3] 266  quiescing auto stats refresher
W221012 19:26:22.305700 2441 jobs/registry.go:824 â‹® [n3] 267  canceling all adopted jobs due to stopper quiescing
W221012 19:26:22.307135 2423 sql/sqlliveness/slinstance/slinstance.go:250 â‹® [n3] 268  exiting heartbeat loop
W221012 19:26:22.307801 1049 jobs/registry.go:824 â‹® [n1] 269  canceling all adopted jobs due to stopper quiescing
W221012 19:26:22.307981 1031 sql/sqlliveness/slinstance/slinstance.go:250 â‹® [n1] 270  exiting heartbeat loop
I221012 19:26:22.308208 1226 jobs/registry.go:1217 â‹® [n1] 271  AUTO SPAN CONFIG RECONCILIATION job 804560481633009665: stepping through state succeeded with error: <nil>
W221012 19:26:22.308299 1911 sql/sqlliveness/slinstance/slinstance.go:250 â‹® [n2] 272  exiting heartbeat loop
W221012 19:26:22.308306 1226 kv/txn.go:705 â‹® [n1] 273  failure aborting transaction: â€¹node unavailable; try another peerâ€º; abort caused by: context canceled
W221012 19:26:22.308609 2819 kv/kvserver/closedts/sidetransport/receiver.go:139 â‹® [n3] 274  closed timestamps side-transport connection dropped from node: 1

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/6911488?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildDeploymentsSection=true&expandBuildTestsSection=true

Jira issue: CRDB-20469

--- a/pkg/sql/sqlstats/persistedsqlstats/datadriven_test.go
+++ b/pkg/sql/sqlstats/persistedsqlstats/datadriven_test.go
@@ -25,6 +25,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlstats"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlstats/persistedsqlstats"
 	"github.com/cockroachdb/cockroach/pkg/sql/tests"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/datapathutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
@@ -33,6 +34,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 	"github.com/cockroachdb/datadriven"
+	"github.com/cockroachdb/errors"
 )
 
 const (
@@ -104,10 +106,13 @@ func TestSQLStatsDataDriven(t *testing.T) {
 		case "exec-sql":
 			stmts := strings.Split(d.Input, "\n")
 			for i := range stmts {
-				_, err := sqlConn.Exec(stmts[i])
-				if err != nil {
-					t.Errorf("failed to execute stmt %s due to %s", stmts[i], err.Error())
-				}
+				testutils.SucceedsSoon(t, func() error {
+					_, exSqlErr := sqlConn.Exec(stmts[i])
+					if exSqlErr != nil {
+						return errors.NewAssertionErrorWithWrappedErrf(exSqlErr, "failed to execute stmt %s", stmts[i])
+					}
+					return nil
+				})
 			}
 		case "observe-sql":
 			actual := observer.QueryStr(t, d.Input)

--- a/pkg/sql/sqlstats/persistedsqlstats/testdata/logical_plan_sampling_for_explicit_txn
+++ b/pkg/sql/sqlstats/persistedsqlstats/testdata/logical_plan_sampling_for_explicit_txn
@@ -1,12 +1,8 @@
-skip issue-num=89861
-----
-
 # This test checks the expected behavior of logical plan sampling.
 # Given a tuple of (db_name, implicitTxn, fingerprint string), the logical plan
 # is only sampled if and only if no logical plan has been sampled for the given
 # tuple in the last 5 minutes. (This is controlled via
 # sql.metrics.statement_details.plan_collection.period cluster setting).
-
 exec-sql
 SET application_name = 'app1'
 ----

